docs/advanced_shaders.rst

```diff
@@ -223,11 +223,39 @@ All fragment functions in Pygfx are somewhat like this:
             @stage(fragment)
             fn fs_main(varyings: Varyings) -> FragmentOutput {
                 ...
-                var out = get_fragment_output(varyings.position, color);
+                var out: FragmentOutput;
+                out.color = vec4<f32>(...);
                 return out;
             }
             """
 
+For some types of blending the output struct is modified automatically,
+and users can influence this process. E.g. to explicitly set a seed for
+dithered blending:
+
+.. code-block::
+
+    ...
+    var out: FragmentOutput;
+    out.color = vec4<f32>(...);
+    $$ if blending == 'dither'
+    out.seed1 = f32(...);
+    $$ endif
+    return out;
+
+... or set the weight for weighted blending:
+
+.. code-block::
+
+    ...
+    var out: FragmentOutput;
+    out.color = vec4<f32>(...);
+    $$ if blending == 'weighted'
+    out.weight = f32(...);
```

Comment:
This isn't exactly consistent with the implementaiton.

In the implementation you use

```
depth_for_weight
```

Your also at the limit of basically creating your own shading language!!!

Reply:
Yeah, that will be fixed in #1002. Not so much create a shading language, but providing entry points for custom shaders, I think.